chain/chain/src/chain.rs

```diff
@@ -2877,6 +2887,28 @@ impl<'a> ChainUpdate<'a> {
         )
     }
 
+    fn get_split_state_roots(
+        &mut self,
+        block: &Block,
+        shard_id: ShardId,
+    ) -> Result<HashMap<ShardUId, StateRoot>, Error> {
+        let next_shard_layout =
+            self.runtime_adapter.get_shard_layout(block.header().next_epoch_id())?;
+        let new_shards = next_shard_layout.get_split_shards(shard_id).expect(&format!(
+            "shard layout must contain maps of all shards to its split shards {}",
+            shard_id
+        ));
+        let state_roots: Result<Vec<_>, _> = new_shards
+            .iter()
+            .map(|shard_uid| {
+                self.chain_store_update
+                    .get_chunk_extra(block.header().prev_hash(), shard_uid)
+                    .map(|chunk_extra| chunk_extra.state_root().clone())
+            })
+            .collect();
+        Ok(new_shards.into_iter().zip(state_roots?.into_iter()).collect())
```

Comment:
consider returning tuple (shard_uid, state_root) from closure and collecting to HashMap to avoid ziping